Add new rule: "SHOULD select appropriate one of date or date-time format" (#808)
zalando/restful-api-guidelines: chapters/data-formats.adoc

more effort to parse, avoid this ambiguity.

[#255]
== {SHOULD} select appropriate one of date or date-time format

When choosing between `date` and `datetime` formats you should take into account the following:

> * `date` should be used for properties that are not time-sensitive or where precise timing is not required,

Reviewer: ```suggestion
* `date` should be used for properties where no exact point in time is not required and day time-range is sufficient,
```

for instance, document dates, birthdays, ETAs (estimated time of arrival).

> Normally date without time implies the time period from midnight to midnight in the local time zone.

Reviewer: ```suggestion
Without further context, `date` without time implies the time period from midnight to midnight in the local time zone.
```

> Since time zone information is absent in the property value, it must be documented and can be also provided

Reviewer: ```suggestion
However, the timezone information can be also provided
```

> as an additional context via other fields indicating location.

Reviewer: ```suggestion
as an additional context information via other fields indicating location.
```

> * `datetime` should be used in all other cases to indicate instants/moments in time.

Reviewer: ```suggestion
* `datetime` should be used in all other cases where an exact point in time is required, 
for instance, datetimes for supplier advice, specific processing events, fast delivery planning dates.
```

> This format requires the explicit time zone offset to be provided,

Reviewer: ```suggestion
As required in <<169>>,  `datetime` requires the explicit time zone offset to be provided,
```
